src/index.ts

```diff
@@ -480,6 +477,11 @@ function startWorkerThread<R, T extends AnyAsyncFn<R>>(
       : []
   ).filter(({ moduleName }) => isPkgAvailable(moduleName))
 
+  // We store a single Byte in the SharedArrayBuffer
+  // for the notification, we can used a fixed size
+  const sharedBuffer = new SharedArrayBuffer(INT32_BYTES)
```

Comment:
Lazy initiation only if they're required. (They're not needed to be initiated when used as worker.)

Reply:
I'm not sure I understand what you mean ?
The SharedArrayBuffer is always needed, it's the mechanism used for `Atomics.wait`

Reply:
oh, I think I understand, you mean it should not be initialized in the worker ?

As it called within `startWorkerThread()` it should only initialize it within the parent thread, not the worker

Reply:
Ah, I was thinking they were global variables, is that possible or has any downside?

Reply:
> but since it resolves synchronously maybe it is not possible for it to have a race condition.

Yep, that's what I was thinking.

Reply:
Hmmm, okay I can do that, I'm just not sure how to initialize this only within the parent thread and not the worker ?

I could put this in a condition (`if (!workerData)`) but then every use of the variable would need to have a conditional check as well. Also it would not work if the function is initially called in a thread (a thread within a thread)

Reply:
```ts
let sharedBuffer: SharedArrayBuffer | undefined

const startWorkerThread = () => {
  sharedBuffer ??= new SharedArrayBuffer()
}
```

Reply:
done